frontend/src/scenes/pipeline/NewButton.tsx

```diff
@@ -1,13 +1,20 @@
+import { useValues } from 'kea'
 import { LemonButton } from 'lib/lemon-ui/LemonButton'
 import { urls } from 'scenes/urls'
 
 import { PipelineStage } from '~/types'
 
+import { pipelineLogic } from './pipelineLogic'
+
 type NewButtonProps = {
     stage: PipelineStage
 }
 
 export function NewButton({ stage }: NewButtonProps): JSX.Element {
+    const { canEnableNewDestinations } = useValues(pipelineLogic)
+    if (stage === PipelineStage.Destination && !canEnableNewDestinations) {
+        return <></>
```

Comment:
Optional nit: You can return null instead (and change the return type of the function to be `| null`)

Reply:
tnx will keep in mind for the future